tests/ci/cdk/pipeline/scripts/build_target.sh

```diff
@@ -0,0 +1,187 @@
+
+  while [[ ${attempt} -le ${MAX_RETRY} ]]; do
+    if [[ $attempt -gt 0 ]]; then
+      echo "Retrying ${attempt}/${MAX_RETRY}..."
+    fi
+
+    attempt=$((attempt + 1))
+
+    echo "Waiting for CI tests for complete. This may take anywhere from 15 minutes to 1 hour"
+    if ! codebuild_build_status_check "${TIMEOUT}"; then
+      echo "Tests failed."
+      if [[ ${attempt} -le ${MAX_RETRY} ]]; then
+        retry_batch_build
+      else
+        echo "CI tests failed."
+        exit 1
+      fi
+    fi
+  done
+
+  echo "All tests completed successfully"
+}
+
+function build_linux_docker_images() {
+  local attempt=0
+
+  echo "Activating AWS CodeBuild to build Linux aarch & x86 docker images."
+  start_codebuild_project aws-lc-docker-image-build-linux "${COMMIT_HASH}"
+
+  while [[ ${attempt} -le ${MAX_RETRY} ]]; do
+    if [[ $attempt -gt 0 ]]; then
+      echo "Retrying ${attempt}/${MAX_RETRY}..."
+    fi
+
+    attempt=$((attempt + 1))
+
+    echo "Waiting for docker images creation. Building the docker images need to take 1 hour."
+    # TODO(CryptoAlg-624): These image build may fail due to the Docker Hub pull limits made on 2020-11-01.
+    if ! codebuild_build_status_check "${TIMEOUT}"; then
+      echo "Build failed."
+      if [[ ${attempt} -le ${MAX_RETRY} ]]; then
+        retry_batch_build
+      else
+        echo "Failed to build Linux docker images"
+        exit 1
+      fi
+    fi
+  done
+
+  echo "Successfully built Linux docker images"
+}
+
+function build_win_docker_images() {
+  local attempt=0
+
+  while [[ ${attempt} -le ${MAX_RETRY} ]]; do
+    if [[ $attempt -gt 0 ]]; then
+      echo "Retrying ${attempt}/${MAX_RETRY}..."
+    fi
+
+    attempt=$((attempt + 1))
+
+    echo "Executing AWS SSM commands to build Windows docker images."
+    if ! start_windows_img_build; then
+      echo "Failed to start build"
+      continue
+    fi
+
+    echo "Waiting for docker images creation. Building the docker images need to take 1 hour."
+    # TODO(CryptoAlg-624): These image build may fail due to the Docker Hub pull limits made on 2020-11-01.
+    if ! win_docker_img_build_status_check "${TIMEOUT}"; then
+      echo "Build failed"
+      continue
+    fi
+
+    echo "Successfully built Windows docker images"
+    exit 0
+  done
+
+  echo "Failed to build Windows docker images"
+  exit 1
+}
+
+while [[ $# -gt 0 ]]; do
+  case ${1} in
+  --build-type)
+    BUILD_TYPE="${2}"
+    shift
+    ;;
+  --platform)
+    PLATFORM="${2}"
+    shift
+    ;;
+  --project)
+    PROJECT="${2}"
+    shift
+    ;;
+  --max-retry)
+    MAX_RETRY="${2}"
+    shift
+    ;;
+  --timeout)
+    TIMEOUT="${2}"
+    shift
+    ;;
+  *)
+    echo "${1} is not supported."
+    exit 1
+    ;;
+  esac
+  shift
+done
+
+MAX_RETRY=${MAX_RETRY:-0}
+TIMEOUT=${TIMEOUT:-180} # 3 hours
+
+if [[ -z ${BUILD_TYPE} ]]; then
+  echo "No build type provided."
+  exit 1
+fi
+
+assume_role
+
+if [[ -z "${BUILD_TYPE+x}" || -z "${BUILD_TYPE}" ]]; then
```

Comment:
```suggestion
if [[ -z "${BUILD_TYPE:+x}" ]]; then
```

Does both of these for you. If BUILD_TYPE is either unset / empty string or not defined it will be a empty string substituted for this check. Otherwise `x` will be substituted and thus not move into the if statement body.